### PR TITLE
catalog-model: update isValidObjectName regex validation

### LINKED ISSUE #9762: Processing user for `github-org` unable to handle user with `-` postfix
## Expected Behavior

Able to process user with `-` postfix

## Actual Behavior

```
[1] 2022-02-23T14:28:53.749Z catalog warn Policy check failed; caused by Error: "metadata.name" is not valid; expected a string that is sequences of [a-zA-Z0-9] separated by any of [-_.], at most 63 characters in total but found "username-". To learn more about catalog file format, visit: https://github.com/backstage/backstage/blob/master/docs/architecture-decisions/adr002-default-catalog-file-format.md type=plugin entity=user:default/username-
```

## Steps to Reproduce

1. Have a user in your orgs with username like `username-` or `whatever-`
2. Set catalog configuration to

```yaml
catalog:
  rules:
    - allow: [Component, API, Group, User, Location, Template]
  locations:
    - type: github-org
      target: https://github.com/myorgs
```

3. You'd receive those warning unable to process user while doing `[1] 2022-02-23T14:28:53.733Z catalog debug Processing user:default/username- type=plugin`

### PR DESCRIPTION
**Pull request by alileza**
## Hey, I just made a Pull Request!

Fixes #9762

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/catalog-model/src/validation/KubernetesValidatorFunctions.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export class KubernetesValidatorFunctions {
       typeof value === 'string' &&
       value.length >= 1 &&
       value.length <= 63 &&
-      /^([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$/.test(value)
+      /^([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9-]$/.test(value)
```
Comment (freben) [inline packages/catalog-model/src/validation/KubernetesValidatorFunctions.ts:52]: you can probably remove the last matcher entirely right?

`^[A-Za-z0-9][-A-Za-z0-9_.]*$`

just on a hunch, untested :)

**Review by freben — COMMENTED** [review]
This needs a changeset too!

File: packages/catalog-model/src/validation/KubernetesValidatorFunctions.ts (hunk 1 above)
Comment (alileza) [reply packages/catalog-model/src/validation/KubernetesValidatorFunctions.ts:52]: hmm, I'm sorry I'm not a regex expert as well, was just checking a few cases with that character and it works :D are you referring to updating this line to your suggestion?

Comment (freben) [reply packages/catalog-model/src/validation/KubernetesValidatorFunctions.ts:52]: Yes. But it seems that the tests fail now, check their output below.

**Pull request closed without merging**